### PR TITLE
bug - non-github links dont work in bounties #1822

### PR DESCRIPTION
**Pull request by alexZielonko**
##### Description

* Allow links to non-github bounties to work from the explorer, including those with query params
* Update link and tooltip text in bounty detail view to use relevant site name, "GitHub" or the repository's hostname
* Avoid invoking gitcoinbot actions for non-github bounties

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

UI

##### Testing

* Expanded test coverage
* Locally confirmed bounty details render for both github and non-github 
* Locally confirmed link/tooltip update as expected

github-detail
non-github-detail

##### Refers/Fixes

Fixes: #1822

### CODE REVIEW

**Comment by alexZielonko** [comment]
I am working on expanding test coverage to support these changes.

**Comment by alexZielonko** [comment]
It looks like 9cb1738 introduces changes causing this pr's new tests to fail. Will investigate further this evening

File: app/app/urls.py [hunk 1]
```diff
@@ -113,6 +113,9 @@
     ),
     re_path(r'^funding/details/?', dashboard.views.bounty_details, name='funding_details'),
 
+    url(r'^issue/?', dashboard.views.bounty_details, name='bounty_details'),
```
Comment (owocki) [inline app/app/urls.py]: whats with the dupe URLS here?

File: app/assets/v2/js/pages/bounty_details.js [hunk 2]
```diff
@@ -759,19 +759,35 @@ var do_actions = function(result) {
     actions.push(_entry);
   }
 
+  function get_repository_name(github_url) {
+    const repo_link = document.createElement('a');
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: why is it necessary to create a new element on the DOM for htis?

Comment (alexZielonko) [reply app/assets/v2/js/pages/bounty_details.js]: This is leveraging the DOM's API to create an object with the attributes of a link, making it straightforward to resolve the site's name for non-github bounties. And as nothing is being inserted into the document, this shouldn't negatively impact client performance. As with most things in JavaScript, there are many ways to address this problem. Can you elaborate more as to your questioning of the necessity of this particular approach?

File: app/app/urls.py (hunk 1 above)
Comment (alexZielonko) [reply app/app/urls.py]: These were originally removed by https://github.com/gitcoinco/web/pull/1887, which prevented non-github bounties from rending. I then committed these changes. Following this, another core member merged a4e146, which is responsible for the duplication you're seeing here. 

I've been trying to keep up with the changes & bugs being introduced into master, but missed the resolution these changes were intended to fix.

**Comment by alexZielonko** [comment]
@owocki 

> this needs testing. does the original bounty that wasn't showing up in the initial PR actually show up on this branch now?

I was able to render both GitHub bounties and those from non-GitHub urls, such as Opengift.io, at the time of the last commit. 

Do you mean this needs testing by me, by another reviewer, or greater test coverage?

**Comment by owocki** [comment]
oh ok, well sounds like you tested it.  then that's good enough for me!

**Comment by alexZielonko** [comment]
I'm working to keep this up to date with master as merge conflicts arise. @owocki is there anything aside from core-contributors' bandwidth keeping this from acceptance?

**Comment by thelostone-mc** [comment]
@owocki @mbeacom ^

File: app/assets/v2/js/pages/bounty_details.js (hunk 2 above)
Comment (owocki) [reply app/assets/v2/js/pages/bounty_details.js]: 👍

File: app/assets/v2/js/pages/bounty_details.js [hunk 3]
```diff
@@ -873,19 +873,35 @@ var do_actions = function(result) {
...
+
+    repo_link.href = github_url;
+
+    const hostname = repo_link.hostname;
+
+    if (hostname.match('github')) {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: is this function pulling the repo name as the function name implies, or the hostname?

File: app/assets/v2/js/pages/bounty_details.js [hunk 4]
```diff
@@ -873,19 +873,35 @@ var do_actions = function(result) {
...
     github_url = github_url.replace('pipermerriam/web3.py', 'ethereum/web3.py');
     github_url = github_url.replace('ethereum/browser-solidity', 'ethereum/remix-ide');
 
+    const repo_name = get_repository_name(github_url);
+
     const _entry = {
       enabled: true,
       href: github_url,
-      text: gettext('View On Github'),
+      text: gettext('View On ' + repo_name),
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: repo name or hostname?

File: app/assets/v2/js/pages/bounty_details.js [hunk 5]
```diff
@@ -1051,27 +1067,44 @@ const build_uri_for_pull_bounty_from_api = function() {
   return uri;
 };
 
+const get_bounty_by_id = function(bountyId) {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: this seems out of scope for this PR... were you running into some issue with looking up by URL (even if it's a non github URL)?

File: app/assets/v2/js/pages/bounty_details.js [hunk 6]
```diff
@@ -1051,27 +1067,44 @@ const build_uri_for_pull_bounty_from_api = function() {
...
+  const deferred = $.Deferred();
+
+  return $.get('/actions/api/v0.1/bounties/' + document.bountyId)
+    .then(result => deferred.resolve([result]));
+};
+
+const get_bounties = function() {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: maybe call it `get_bounty` instead of `get_bounties` since it'lll really only ever be expected to return one?

**Comment by owocki** [comment]
sorry guys... just responded here...


cc @mbeacom in case he wants to review

**Comment by alexZielonko** [comment]
@owocki I've made local changes to reflect those suggested by your review comments. It appears changes to upstream/master are affecting functionality introduced by this pull request. I will resolve these before committing and post an additional comment when this is complete. I had a bit more free time when I initially took on this bounty, but hope to get to it within the next week. 

Re: 

> what kind of testing did you do @alexZielonko !?

I have expanded test coverage and am confirming non-github & GitHub bounty details render as expected locally with each commit.

**Comment by alexZielonko** [comment]
I have been unable to hop back over to this project and can't prioritize this again for the foreseeable future. Closing pull request, best of luck

**Comment by vs77bb** [comment]
@owocki prolly can claw back funds on the bounty here, yeah?
